### DISCUSSION TITLE
Is there a way to set global options?

### DISCUSSION THREAD

**Discussion by abdulhdr1** [47ng/nuqs#635] [Q&A] [answered]
I don't want to pass `{ clearOnDefault: true }` for every `useQueryState` call
I know I can abstract this out with a hook, just curious if there's a way to set it within the lib, didn't find anything in the docs

**Comment by franky47** [comment]
There isn't a way to do that currently, but that would be nice.

Feel free to open a PR!

**Reply by abdulhdr1** [reply]
Sure!

Any idea what the API should look like? 

1. Create a Context, like react-query
2. File-based, extend `package.json` (like [prettier](https://prettier.io/docs/en/configuration.html)) or a custom `nuqs.config.ts` (like [eslint](https://eslint.org/docs/latest/use/configure/configuration-files))
3. Expose a function for doing so `import { setGlobalOptions } from 'nuqs'` that the user can run on startup

**Reply by franky47** [reply]
1. I like this idea, it would allow users to scope different defaults to different parts of their apps. May be tricky to implement and consume though.
2. Definitely not, we have too many config files for everything, no need to add to the pile if we can avoid it.
3. I initially thought about this, that might be the easiest to implement, with a top-level object holding the default values that get consumed in the hooks. The only catch is to call this in a `'use client'` context.

**Reply by franky47** [reply]
Let's see what the community thinks about it: https://x.com/fortysevenfx/status/1833453507673837810

**Reply by franky47** [reply, marked as answer]
Note: `clearOnDefault` is now true by default in v2.

I've been thinking about adding a GlobalDefault provider, but it would only work in the hooks, and not in the cache or the serializer, so that's unlikely to land anytime soon.
